fda/depth_measures.py

```diff
@@ -0,0 +1,388 @@
+    """Implementation of Modified Band Depth for functional data.
+
+    The band depth of each sample is obtained by computing the fraction of time its graph is contained
+    in the bands determined by two sample curves. In the case the fdatagrid domain dimension is 2, instead
+    of curves, surfaces determine the bands.
+
+    Args:
+        fdatagrid (FDataGrid): Object over whose samples the modified band depth is going to be calculated.
+        pointwise (boolean, optional): Indicates if the pointwise depth is also returned. Defaults to False.
+
+    Returns:
+        depth (numpy.darray): Array containing the modified band depth of the samples.
+
+    Returns:
+        depth_pointwise (numpy.darray, optional): Array containing the modified band depth of
+        the samples at each point of discretisation. Only returned if pointwise equals to True.
+
+    Examples:
+        Univariate setting specifying pointwise:
+
+        >>> data_matrix = [[1, 1, 2, 3, 2.5, 2], [0.5, 0.5, 1, 2, 1.5, 1],
+        ...                [-1, -1, -0.5, 1, 1, 0.5], [-0.5, -0.5, -0.5, -1, -1, -1]]
+        >>> sample_points = [0, 2, 4, 6, 8, 10]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> modified_band_depth(fd, pointwise = True)
+        (array([[0.5       ],
+               [0.83333333],
+               [0.72222222],
+               [0.66666667]]), array([[[0.5       ],
+                [0.5       ],
+                [0.5       ],
+                [0.5       ],
+                [0.5       ],
+                [0.5       ]],
+        <BLANKLINE>
+               [[0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333]],
+        <BLANKLINE>
+               [[0.5       ],
+                [0.5       ],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333]],
+        <BLANKLINE>
+               [[0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.5       ],
+                [0.5       ],
+                [0.5       ]]]))
+
+        Multivariate Setting without specifying pointwise:
+
+        >>> data_matrix = [[[[1, 3], [2, 6]], [[23, 54], [43, 76]], [[2, 45], [12, 65]]],
+        ...                [[[21, 34], [8, 16]], [[67, 43], [32, 21]], [[10, 24], [3, 12]]],
+        ...                [[[4, 6], [4, 10]], [[45, 48], [38, 56]], [[34, 78], [10, 28]]]]
+        >>> sample_points = [[2, 4, 6], [3, 6]]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> modified_band_depth(fd)
+        array([[0.66666667, 0.72222222],
+               [0.72222222, 0.66666667],
+               [0.94444444, 0.94444444]])
+
+    """
+    if fdatagrid.ndim_domain > 2:
+        raise NotImplementedError("Only support 1 or 2 dimensions on the domain.")
+
+    if fdatagrid.ndim_domain == 1:
```

Comment:
Can the same code be used for all dimension numbers?

Reply:
implemented